explorer/mempool.go

```diff
@@ -180,23 +183,30 @@ func (exp *explorerUI) storeMempoolInfo() (lastBlockHash string, lastBlock int64
 
 	var totalOut float64
 	var totalSize int32
+	var votingInfo VotingInfo
+
+	lastBlockHash, lastBlock, lastBlockTime = exp.getLastBlock()
+
+	votingInfo.TotalNeeded = exp.ChainParams.TicketsPerBlock
+	votingInfo.Required = exp.ChainParams.StakeRewardProportion
 
-	// Categorize the transactions, and bin votes by ticket spent
-	txindexes := make(map[string]int)
+	votingInfo.voted = make(map[string]bool)
+
+	txindexes := make(map[int64]map[string]int)
 	for _, tx := range memtxs {
 		switch tx.Type {
 		case "Ticket":
 			tickets = append(tickets, tx)
 		case "Vote":
-			if !voteForLastBlock(lastBlockHash, tx.VoteInfo.Validation.Hash) {
-				continue
-			}
-			if idx, ok := txindexes[tx.VoteInfo.TicketSpent]; ok {
-				tx.VoteInfo.MempoolTicketIndex = idx
-			} else {
-				idx := len(txindexes) + 1
-				txindexes[tx.VoteInfo.TicketSpent] = idx
-				tx.VoteInfo.MempoolTicketIndex = idx
+			addVoteIndex(tx.VoteInfo, txindexes)
+			if tx.VoteInfo.ForLastBlock = voteForLastBlock(lastBlockHash, tx.VoteInfo.Validation.Hash); tx.VoteInfo.ForLastBlock && !votingInfo.voted[tx.VoteInfo.TicketSpent] {
```

Comment:
please put the assignment on a separate line

Reply:
done